ringpop.go

```diff
@@ -346,14 +346,23 @@ func (rp *Ringpop) Bootstrap(userBootstrapOpts *swim.BootstrapOptions) ([]string
 	if err != nil {
 		return nil, err
 	}
+	hosts, err := userbootstrapOpts.DiscoverProvider.Hosts()
+	if err != nil {
+		rp.logger.WithField("error", err).Info("Failed to get boostrap hosts")
+		return nil, err
+	}
 
-	// If the user has provided a list of hosts (and not a bootstrap file),
-	// check we're in the bootstrap host list and add ourselves if we're not
+	// Check we're in the bootstrap host list and add ourselves if we're not
 	// there. If the host list is empty, this will create a single-node
 	// cluster.
-	bootstrapOpts := *userBootstrapOpts
-	if len(bootstrapOpts.File) == 0 && !util.StringInSlice(bootstrapOpts.Hosts, identity) {
-		bootstrapOpts.Hosts = append(bootstrapOpts.Hosts, identity)
+	// TODO: move this code to the caller of the Hosts().
+	bootstrapOpts := *userbootstrapOpts
+	switch bootstrapOpts.DiscoverProvider.(type) {
```

Comment:
That was the idea to first post a compatible change, and later, in a separate PR, fix that (see the PR message).

I will send an update with this section fixed.